src/main/java/Duke.java

```diff
@@ -1,10 +1,223 @@
+        String[] userInputArray = userInput.split(USER_INPUT_SPLITTER);
+        int deleteTaskNumber = Integer.parseInt(userInputArray[1]);
+        int indexOfTaskToDelete = deleteTaskNumber - 1;
+
+        Task taskToDelete = taskArrayList.get(indexOfTaskToDelete);
+        taskArrayList.remove(indexOfTaskToDelete);
+        deleteTaskMessage(taskToDelete);
+        printNumberOfTask();
+    }
+
+    public static void printInvalidTaskNumberProvided() {
+        System.out.println("Task number provided does not exist!\n");
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        welcomeMessage();
+        Scanner scanner = new Scanner(System.in);
+        Pattern doneTaskPattern = Pattern.compile(DONE_TASK_INDICATOR);
+        Pattern deleteTaskPattern = Pattern.compile(DELETE_TASK_INDICATOR);
+        DukeStorage dukeTextFile = new DukeStorage();
+        taskArrayList = dukeTextFile.getTaskArrayList();
+
+        while (true) {
+            String userInput = scanner.nextLine();
+            Matcher doneMatcher = doneTaskPattern.matcher(userInput);
+            boolean taskIsDone = doneMatcher.find();
+
+            Matcher deleteMatcher = deleteTaskPattern.matcher(userInput);
+            boolean deleteTask = deleteMatcher.find();
+
+            if (userInput.equals(BYE_INDICATOR)) {
+                break;
+            } else if (userInput.equals(LIST_INDICATOR)) {
+                getAllTask(taskArrayList);
+            } else if (taskIsDone) {
+                try {
+                    markTaskAsDone(userInput, taskArrayList);
+                    dukeTextFile.saveToDukeTextFile(taskArrayList);
+                } catch (IndexOutOfBoundsException e) {
+                    printInvalidTaskNumberProvided();
+                }
+            } else if (deleteTask) {
+                try {
+                    deleteTask(userInput, taskArrayList);
+                    dukeTextFile.saveToDukeTextFile(taskArrayList);
+                } catch (IndexOutOfBoundsException e) {
+                    printInvalidTaskNumberProvided();
+                }
+            } else {
+                try {
+                    addTaskToTaskArray(userInput, taskArrayList);
+                    dukeTextFile.saveToDukeTextFile(taskArrayList);
+                } catch (EmptyDescriptionException e) {
+                    e.printExceptionMessage();
+                } catch (InvalidTaskTypeException e) {
+                    e.printExceptionMessage();
+                } catch (InvalidDateIndicatorException e) {
+                    e.printExceptionMessage();
+                }
+            }
```

Comment:
The condition `userInput.equals(BYE_INDICATOR))` is not as meaningful as `deleteTask`. How about assigning it a [name](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain)?